corehq/apps/locations/tests/test_api_v6.py

```diff
@@ -0,0 +1,111 @@
+import json
+from corehq.apps.api.tests.utils import APIResourceTest
+from corehq.apps.locations.models import LocationType, SQLLocation
+from corehq.apps.locations.resources.v0_6 import LocationResource
+from corehq.util.view_utils import absolute_reverse
+
+
+class LocationV6Test(APIResourceTest):
+    api_name = 'v0.6'
+    resource = LocationResource
+
+    def setUp(self):
+        self.parent_type = LocationType.objects.create(
+            domain=self.domain.name,
+            name="State",
+            code="state",
+        )
+
+        self.child_type = LocationType.objects.create(
+            domain=self.domain.name,
+            name="City",
+            code="city",
+            parent_type=self.parent_type
+        )
+
+        self.location1 = SQLLocation.objects.create(
+            domain=self.domain.name,
+            id="1234",
+            location_id="1",
+            name="Colorado",
+            site_code="colorado",
+            location_type=self.parent_type
+        )
+
+        self.location2 = SQLLocation.objects.create(
+            domain=self.domain.name,
+            id="1235",
+            location_id="2",
+            name="Denver",
+            site_code="denver",
+            longitude=10.1234567891,
+            latitude=11.1234567891,
+            location_type=self.child_type,
+            parent=self.location1,
+            metadata={"population": "715,522"}
+        )
+
+    def single_endpoint(self, id):
+        return absolute_reverse('api_dispatch_detail', kwargs=dict(domain=self.domain.name,
+                                                          api_name=self.api_name,
+                                                          resource_name=self.resource._meta.resource_name,
+                                                          location_id=id))
+
+    # Disregards order of fields in output.
+    def _validate_obj_output(self, desired_output, actual_output):
```

Comment:
You can also use `assertDictEqual`, which'll also ignore order.  (or even `assertEqual`, which just calls `assertDictEqual` if both items are dicts).  In googling to verify this, I found out that [dict equality _always_ disregards ordering](https://docs.python.org/3/library/stdtypes.html#mapping-types-dict), presumably because they used to be unordered.

> Dictionaries compare equal if and only if they have the same (key, value) pairs (regardless of ordering).

Reply:
That should be fine, I don't think order is considered significant in JSON anyways, so it's likely to get lost at some point anyways.